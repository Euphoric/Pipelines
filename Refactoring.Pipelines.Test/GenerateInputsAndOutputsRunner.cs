using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ApprovalTests;
using ApprovalTests.Core;
using ApprovalTests.Reporters;

namespace Refactoring.Pipelines.Test
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class GenerateInputsAndOutputsRunner
    {
        [TestMethod]
        public void GenerateInputsAndOutputs()
        {
            var inputCount = 2;
            var outputCount = 2;
            var subject = new InputsAndOutputsGenerator(inputCount, outputCount);
            Approvals.Verify(subject);
        }

        [TestMethod]
        public void GenerateInputsExtensions()
        {
            var inputCount = 2;
            var subject = new InputsExtensionsGenerator(inputCount);
            Approvals.Verify(subject);
        }

        [TestMethod]
        public void GenerateInputs()
        {
            var inputCount = 1;
            var outputCounts = new[] {1, 2};
            var subject = new InputsGenerator(inputCount, outputCounts);
            Approvals.Verify(subject);
        }
    }
}
