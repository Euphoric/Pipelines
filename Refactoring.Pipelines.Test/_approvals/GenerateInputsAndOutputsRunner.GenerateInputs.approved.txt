namespace Refactoring.Pipelines.InputsAndOutputs
{
    public class Inputs<TInput1>
    {
        private readonly IGraphNode _node;
        public Inputs(IGraphNode node) { this._node = node; }

        public Inputs1AndOutputs1<TInput1, TOutput1> AndOutputs<TOutput1>()
        {
            return new Inputs1AndOutputs1<TInput1, TOutput1>(this._node);
        }

        public Inputs1AndOutputs2<TInput1, TOutput1, TOutput2> AndOutputs<TOutput1, TOutput2>()
        {
            return new Inputs1AndOutputs2<TInput1, TOutput1, TOutput2>(this._node);
        }

    }
}
